### PR TITLE
Disable terser on polymer bundle in dev assets mode

### PR DESCRIPTION
**Pull request by psybuzz**
While the Angular bundle incremental build time was improved
with the `tensorboard:dev` mode, the Polymer 3 bundle still
relies on Rollup. After Terser was introduced in the build, the
Polymer incremental build time can take 30+ sec on top of the
TypeScript compilation time.

This change removes the Terser step from the Polymer bundling
phase, only for the dev assets mode.

### CODE REVIEW

**Review by stephanwlee — COMMENTED** [review]
Would it make more sense to use the tf_dev_js_binary on the polymer bundle?

**Comment by psybuzz** [comment]
> Would it make more sense to use the tf_dev_js_binary on the polymer bundle?

Ideally yes, and I find the PR as-is quite unfortunate.  Sadly, I couldn't find any UMD-style bundle of Polymer (the library) itself.  Without that, `@polymer/polymer` imports from our components won't be resolved without the Rollup step first.

File: tensorboard/components/BUILD [hunk 1]
```diff
@@ -1,4 +1,4 @@
-load("//tensorboard/defs:defs.bzl", "tf_js_binary", "tf_ts_library")
+load("//tensorboard/defs:defs.bzl", "tf_dev_js_binary", "tf_js_binary", "tf_ts_library")
```
Comment (stephanwlee) [inline tensorboard/components/BUILD:1]: remove unused dev_js_binary.

File: tensorboard/components/BUILD [hunk 2]
```diff
@@ -38,13 +38,33 @@ tf_ts_library(
...
 
+# A target for development only. By dropping 'compile', this skips the expensive
+# minification step.
+tf_js_binary(
+    name = "polymer3_lib_binary_no_shim_dev",
+    compile = False,
+    entry_point = ":polymer3_lib.ts",
+    deps = [":polymer3_ts_lib"],
+)
+
```
Comment (stephanwlee) [inline tensorboard/components/BUILD:49]: This might belong in `tensorboard/webapp/dev_assets` maybe?

**Review by stephanwlee — COMMENTED** [review]
Alternative is to build `@polymer/polymer` using rollup but we need to export polymer symbols outside of that bundle.

**Review by stephanwlee — APPROVED** [review]

File: tensorboard/components/BUILD (hunk 1 above)
Comment (psybuzz) [reply tensorboard/components/BUILD]: Done.

File: tensorboard/components/BUILD (hunk 2 above)
Comment (psybuzz) [reply tensorboard/components/BUILD]: Done.

**Review by psybuzz — COMMENTED** [review]
Thanks for the review!

**Pull request merged**